app/services/next_course_creation_step_service.rb

```diff
@@ -1,48 +1,64 @@
 class NextCourseCreationStepService
-  def execute(current_step:, current_provider:)
-    case current_step
-    when :level
-      :subjects
-    when :subjects
-      :age_range
-    when :age_range
-      :outcome
-    when :outcome
-      handle_outcome(current_provider)
-    when :fee_or_salary
-      :full_or_part_time
-    when :apprenticeship
-      :full_or_part_time
-    when :full_or_part_time
-      :location
-    when :location
-      handle_location(current_provider)
-    when :accredited_body
-      :entry_requirements
-    when :entry_requirements
-      :applications_open
-    when :applications_open
-      :start_date
-    when :start_date
-      :confirmation
-    end
+  def execute(current_step:, current_provider:, level:)
+    get_next_step(get_workflow(current_provider, level), current_step)
   end
 
-private
+  def get_next_step(steps, current_step)
+    steps[steps.find_index(current_step).next]
+  end
 
-  def handle_outcome(provider)
-    if provider.accredited_body?
-      :apprenticeship
-    else
-      :fee_or_salary
+  def get_workflow(provider, level)
+    if is_further_education?(level)
+      %i[
+        level
+        outcome
+        full_or_part_time
+        location
+        applications_open
+        start_date
+        confirmation
+      ]
+    elsif is_uni_or_scitt?(provider)
+      %i[
+        level
+        subjects
+        age_range
+        outcome
+        apprenticeship
+        full_or_part_time
+        location
+        entry_requirements
+        applications_open
+        start_date
+        confirmation
+      ]
+    elsif is_school_direct?(provider, level)
+      %i[
+        level
+        subjects
+        age_range
+        outcome
+        fee_or_salary
+        full_or_part_time
+        location
+        accredited_body
+        entry_requirements
+        applications_open
+        start_date
+        confirmation
+      ]
```

Comment:
Not that I think we should jump on every one of CodeClimate's concerns, but these lists seem like they could very nicely be broken up into methods with `get_workflow` concerned with which one to return.